amundsen_application/static/js/ducks/search/api/v0.ts

```diff
@@ -1,24 +1,36 @@
-import axios, { AxiosResponse, AxiosError } from 'axios';
+import axios, { AxiosError, AxiosResponse } from 'axios';
 
-import { SearchResponse } from '../types';
+import { SearchAllRequest, SearchResponse, SearchResourceRequest } from '../types';
 
-import { SearchReducerState } from '../reducer';
-
-function transformSearchResults(data: SearchResponse): SearchReducerState {
-  return {
-    searchTerm: data.search_term,
-    dashboards: data.dashboards,
-    tables: data.tables,
-    users: data.users,
-  };
+export function searchAll(action: SearchAllRequest) {
+  const { term, options } = action;
+  let baseUrl = '/api/search/v0';
```

Comment:
nit: can make `'/api/search/v0'` a constant. Not a big deal here, but useful if we had a lot of methods and want to update the base url, we'd just have to do it in one place.